tests/test_package_index.py

```diff
@@ -0,0 +1,72 @@
+import json
+from pathlib import Path
+
+import pytest
+
+import micropip.package_index as package_index
+
+TEST_TEMPLATES_DIR = Path(__file__).parent / "test_data" / "pypi_response"
+
+
+@pytest.mark.parametrize("name", ["black", "pytest", "snowballstemmer"])
+def test_project_info_from_json(name):
+    test_file = TEST_TEMPLATES_DIR / f"{name}_json.json"
```

Comment:
Can you gzip the json with data, so that grepping the source code wouldn't produce all sorts of noise with text data?

Reply:
Good point! Will do.